config/config.go

```diff
@@ -68,6 +68,99 @@ func NewCloudConfig(contents string) (*CloudConfig, error) {
 	return &cfg, err
 }
 
+func (co *CoreOS) Merge(mergeWith CoreOS) {
```

Comment:
system is probably the most appropriate package for the implementation. It will have to be called from initialize.Apply, like you alluded. I like the idea of applying configs serially. That fits with the rest of the architecture and simplifies the code.